### ISSUE TITLE
support css variables to propertie

### ISSUE THREAD

**Issue by tamascsaba** [sasstools/sass-lint#877]
If I use css variables:
```
.element {
  --main-bg-color: brown;
}
```

sass-lint throw: no-misspelled-properties

**Comment by eugef** [comment]
With the version 1.0 I am having following error when using css variables:

```
413:1  error  Please check validity of the block starting from line #413  Fatal
```

**Comment by dimanech** [comment]
Have the same issue. Since CSS custom property not equal to Sass variables we have too many reasons to have both in sass file.

**Comment by DanPurdy** [comment]
Hi guys, you need to report this to [Gonzales-pe](https://github.com/tonyganch/gonzales-pe) which is the AST we use.

The fatal errors you're encountering are because the AST doesn't currently support CSS variables / modules

**Comment by eugef** [comment]
Seems like this issue is related https://github.com/tonyganch/gonzales-pe/issues/197

**Comment by HHogg** [comment]
If this is any help to anyone, I've managed to get around this for the moment by changing

```scss
:root {
  --custom-var: white;
}
```
to
```scss
:root {
  #{--custom-var}: white;
}
```

Not great but it suppresses the error thrown for the time being. I've not found any issues _yet_.

**Comment by Rjervis3** [comment]
@HHogg , Thanks, [your solution](https://github.com/sasstools/sass-lint/issues/877#issuecomment-267176435) works!

**Comment by robsterlini** [comment]
@HHogg, have you found a way to handle multiple vars within a single block? I get a `no-duplicate-properties` error thrown by the linter with:

```Scss
.foo {
  #{--bar}: red;
  #{--baz}: blue;
}
```

**Comment by HHogg** [comment]
Sorry @robsterlini I cant remember. I stopped using Sass a couple of years ago, and swapped to PostCSS/CSS Next.

**Comment by alexmccabe** [comment]
Is there any progress on this?

**Comment by Snugug** [comment]
Thanks for your patience all; we're looking to onboard additional maintainers this week who will be looking at all of the outstanding PRs

**Comment by shennan** [comment]
Would it be possible to get an update on this progress of this issue?

**Comment by nelson6e65** [comment]
Seems that this (https://github.com/tonyganch/gonzales-pe/issues/290) had been resolved by https://github.com/tonyganch/gonzales-pe/pull/295

**Comment by Stephane-Pires** [comment]
@robsterlini Have you find a fix for no-duplicate-properties on css variables 

i'm still getting the no-duplicate-properties error

**Comment by robsterlini** [comment]
> @robsterlini Have you find a fix for no-duplicate-properties on css variables
> 
> i'm still getting the no-duplicate-properties error

Sadly not, I have been using `// sass-lint:disable` to force those rules to be ignored in those instances.
